File: Python Language/Q7.py
# Program to calculate Total and Average marks

# Getting Roll Number
roll_no = input("Enter Student Roll No: ")

# Getting marks for three subjects
prog_fund = float(input("Enter marks for Programming Fundamentals: "))
intro_comp = float(input("Enter marks for Intro to Computing: "))
comp_graph = float(input("Enter marks for Computer Graphics: "))

# Calculating Total
total_marks = prog_fund + intro_comp + comp_graph

# Calculating Average (Total divided by 3)
average_marks = total_marks / 3

# Printing the Report
print("\n--- Student Result ---")
print(f"Roll No: {roll_no}")
print(f"Total Marks: {total_marks}")
print(f"Average Marks: {average_marks:.2f}") # .2f rounds to 2 decimal places

File: C++ Language/Q7.cpp
#include <iostream>
using namespace std;

int main() {
    int rollNo;
    float marksProg, marksComp, marksGraph;
    float total, average;

    // Getting student details
    cout << "Enter Roll No: ";
    cin >> rollNo;

    cout << "Enter marks for Programming Fundamentals: ";
    cin >> marksProg;

    cout << "Enter marks for Intro to Computing: ";
    cin >> marksComp;

    cout << "Enter marks for Computer Graphics: ";
    cin >> marksGraph;

    // Calculating Total
    total = marksProg + marksComp + marksGraph;

    // Calculating Average (divide by 3.0 to keep decimal precision)
    average = total / 3.0;

    cout << "-----------------------" << endl;
    cout << "Roll No: " << rollNo << endl;
    cout << "Total Marks: " << total << endl;
    cout << "Average Marks: " << average << endl;

    return 0;
}